diff --git a/node_modules/esbuild-wasm/lib/main.js b/node_modules/esbuild-wasm/lib/main.js
index 8e02ab5..3e7ac7d 100644
--- a/node_modules/esbuild-wasm/lib/main.js
+++ b/node_modules/esbuild-wasm/lib/main.js
@@ -1926,7 +1926,7 @@ var ensureServiceIsRunning = () => {
   let [command, args] = esbuildCommandAndArgs();
   let child = child_process.spawn(command, args.concat(`--service=${"0.18.5"}`, "--ping"), {
     windowsHide: true,
-    stdio: ["pipe", "pipe", "inherit"],
+    stdio: ["pipe", "pipe", "pipe"],
     cwd: defaultWD
   });
   let { readFromStdout, afterClose, service } = createChannel({
@@ -2092,7 +2092,7 @@ error: ${text}`);
       throw new Error(`Internal error: Expected id ${id} but got id ${id2}`);
     if (reject) {
       applyProperties(reject, properties);
-      throw reject;
+      return fakeBuildError(reject);
     }
     return resolve;
   };
