diff --git a/node_modules/jsii/lib/compiler.js b/node_modules/jsii/lib/compiler.js
index 8b94573..65529ee 100644
--- a/node_modules/jsii/lib/compiler.js
+++ b/node_modules/jsii/lib/compiler.js
@@ -17,7 +17,8 @@ const BASE_COMPILER_OPTIONS = {
     declaration: true,
     experimentalDecorators: true,
     incremental: true,
-    lib: ['lib.es2020.d.ts'],
+    lib: ['lib.es2020.d.ts','lib.dom.d.ts'],
+    esModuleInterop: true,
     module: ts.ModuleKind.CommonJS,
     newLine: ts.NewLineKind.LineFeed,
     noEmitOnError: true,
