diff --git a/lib/main.js b/lib/main.js
index 3001b4dca248b9786aa35b5cbd9589f927758c6d..9cd981adcb53de28d68fcf9d0a1e21a12a168f64 100644
--- a/lib/main.js
+++ b/lib/main.js
@@ -1940,7 +1940,7 @@ var ensureServiceIsRunning = () => {
   let [command, args] = esbuildCommandAndArgs();
   let child = child_process.spawn(command, args.concat(`--service=${"0.18.20"}`, "--ping"), {
     windowsHide: true,
-    stdio: ["pipe", "pipe", "inherit"],
+    stdio: ["pipe", "pipe", "pipe"],
     cwd: defaultWD
   });
   let { readFromStdout, afterClose, service } = createChannel({
@@ -2106,7 +2106,7 @@ error: ${text}`);
       throw new Error(`Internal error: Expected id ${id} but got id ${id2}`);
     if (reject) {
       applyProperties(reject, properties);
-      throw reject;
+      return fakeBuildError(reject);
     }
     return resolve;
   };