diff --git a/index.js b/index.js
index 89afd7e584a50233d6948255c4a4f52edbaf297c..d644ff4def660d58c6a122816b25c9cce22cd8eb 100644
--- a/index.js
+++ b/index.js
@@ -21,7 +21,22 @@ var escapeHtml = require('escape-html')
 var etag = require('etag')
 var fresh = require('fresh')
 var fs = require('fs')
-var mime = require('mime')
+var newMime = require('mime')
+
+
+// patch in mimeV1 support
+var oldMime = {
+  ...newMime,
+  lookup: newMime.getType,
+  default_type: newMime.getType('bin'),
+  charsets: {
+    lookup: function(mimeType, fallback) {
+      // Assume text types are utf8
+      return (/^text\/|^application\/(javascript|json)/).test(mimeType) ? 'UTF-8' : fallback;
+    }
+  }
+}
+
 var ms = require('ms')
 var onFinished = require('on-finished')
 var parseRange = require('range-parser')
@@ -68,7 +83,7 @@ var UP_PATH_REGEXP = /(?:^|[\\/])\.\.(?:[\\/]|$)/
  */
 
 module.exports = send
-module.exports.mime = mime
+module.exports.mime = oldMime;
 
 /**
  * Return a `SendStream` for `req` and `path`.
@@ -827,14 +842,14 @@ SendStream.prototype.type = function type (path) {
 
   if (res.getHeader('Content-Type')) return
 
-  var type = mime.lookup(path)
+  var type = oldMime.lookup(path)
 
   if (!type) {
     debug('no content-type')
     return
   }
 
-  var charset = mime.charsets.lookup(type)
+  var charset = oldMime.charsets.lookup(type)
 
   debug('content-type %s', type)
   res.setHeader('Content-Type', type + (charset ? '; charset=' + charset : ''))
diff --git a/package.json b/package.json
index 7f269d515c8bf4d94510578600a9bf9fc9f43fd3..ccf3d5c6626ec87fc7a79ce5a631b7d8e35dddea 100644
--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
     "etag": "~1.8.1",
     "fresh": "0.5.2",
     "http-errors": "2.0.0",
-    "mime": "1.6.0",
+    "mime": "3.0.0",
     "ms": "2.1.3",
     "on-finished": "2.4.1",
     "range-parser": "~1.2.1",